sumy/nlp/tokenizers.py

```diff
@@ -77,6 +77,23 @@ def tokenize(self, text):
         return kkma.nouns(text)
 
 
+class GreekSentencesTokenizer:
+    """Calls sent_tokenize for greek, which doesn't split sentences on semicolon ';' 
+    or the greek question mark ';'. The regex below splits on both.
+    """
+    @classmethod
+    def tokenize(self, text):
+        sentences = nltk.sent_tokenize(text, language = 'greek')
+        sentences = [list(filter(None, re.split(r'(?<=[;;])\s+', sentence))) for sentence in sentences]
```

Comment:
Because you are using these collections in the for loop below just once I think they can be generators.
```suggestion
        sentences = (filter(None, re.split(r'(?<=[;;])\s+', sentence)) for sentence in sentences)
```